File: output/test_04.py
import requests
import requests_mock

def test_api():
    with requests_mock.Mocker() as mocker:
        mocker.get('http://localhost:3000/api/users', status_code=200, json=["user1", "user2"])
        mocker.post('http://localhost:3000/api/users', status_code=201, json=["user3"])
        response = requests.get('http://localhost:3000/api/users')
        assert response.status_code == 200
        assert all(item in response.json() for item in ["user1", "user2"])

        response = requests.post('http://localhost:3000/api/users', json={"user": {"name": "user3", "email": "user3@example.com"}})
        assert response.status_code == 201
        assert all(item in response.json() for item in ["user3"])

File: input/test_04.rb
require 'rails_helper'

RSpec.describe 'API', type: :request do
  it 'returns a list of users' do
    get '/api/users'
    expect(response).to have_http_status(200)
    expect(response.body).to include('user1', 'user2')
  end

  it 'creates a user' do
    post '/api/users', params: { user: { name: 'user3', email: 'user3@example.com' } }
    expect(response).to have_http_status(201)
    expect(response.body).to include('user3')
  end
end
